aioshelly/common.py

```diff
@@ -137,3 +139,29 @@ def shelly_supported_firmware(result: dict[str, Any]) -> bool:
     # We compare firmware release dates because Shelly version numbering is
     # inconsistent, sometimes the word is used as the version number.
     return int(match[0]) >= fw_ver
+
+
+async def trigger_ota_http(
+    session: aiohttp.ClientSession,
+    host: str,
+    gen: int,
+    auth: aiohttp.BasicAuth | None = None,
+) -> bool:
+    """Trigger a firmware update via OTA http endpoint."""
+    if gen in BLOCK_GENERATIONS:
+        path = "ota"
+    else:
+        path = "rpc/Shelly.Update"
+    try:
+        await session.get(
+            URL.build(scheme="http", host=host, path=f"/{path}"),
+            auth=auth,
+            timeout=HTTP_CALL_TIMEOUT,
+        )
+    except aiohttp.ClientResponseError:
+        return False
+    except CONNECT_ERRORS:
+        return False
+
+    _LOGGER.debug("Update in progress for device %s", host)
+    return True
```

Comment:
It might be a bit nicer to throw a named exception so we can provide a better error in the UI if the update fails for some reason